### PR TITLE
docs(upgrading/remix): update with latest info

### PR DESCRIPTION
**Pull request by brookslybrand**

### CODE REVIEW

File: docs/upgrading/remix.md [hunk 1]
```diff
@@ -45,32 +56,37 @@ npm install
...
+| ---------------------------------- | --- | -------------------------------------------- |
+| `@remix-run/architect`             | ➡️  | `@react-router/architect`                    |
+| `@remix-run/cloudflare`            | ➡️  | `@react-router/cloudflare`                   |
+| `@remix-run/dev`                   | ➡️  | `@react-router/dev`                          |
+| `@remix-run/express`               | ➡️  | `@react-router/express`                      |
+| `@remix-run/fs-routes`             | ➡️  | `@react-router/fs-routes`                    |
+| `@remix-run/node`                  | ➡️  | `@react-router/node`                         |
+| `@remix-run/react`                 | ➡️  | `react-router`                               |
+| `@remix-run/route-config`          | ➡️  | `@react-router/dev`                          |
+| `@remix-run/routes-option-adapter` | ➡️  | ` @react-router/remix-routes-option-adapter` |
```
Comment (markdalgleish) [inline docs/upgrading/remix.md:74]:
```suggestion
| `@remix-run/routes-option-adapter` | ➡️  | `@react-router/remix-routes-option-adapter`  |
```

File: docs/upgrading/remix.md [hunk 2]
```diff
@@ -150,7 +281,7 @@ However, the short-term solution to ease the upgrade is to use TypeScript's [mod
 
 You can do this with the following code in your `vite.config.ts`:
```
Comment (markdalgleish) [inline docs/upgrading/remix.md:282]:
```suggestion
You can do this with the following code in `react-router.config.ts`:
```

Comment (pcattori) [reply docs/upgrading/remix.md:282]: We need to make sure that whatever file is used for `interface AppLoadContext` is included in your `tsconfig.json`'s `include`. I'm guessing most people will have `**/*.ts` which will cover this, but might be worth mentioning.

File: docs/upgrading/remix.md [hunk 3]
```diff
@@ -79,9 +95,127 @@ Update the scripts in your `package.json`:
...
+-import { type RouteConfig } from "@remix-run/route-config";
+-import { flatRoutes } from "@remix-run/fs-routes";
+-import { remixRoutesOptionAdapter } from "@remix-run/routes-option-adapter";
++import { type RouteConfig } from "@react-router/dev/routes";
++import { flatRoutes } from "@react-router/fs-routes";
++import { remixRoutesOptionAdapter } from "@react-router/remix-routes-option-adapter";
+
+export default {
+  // however your routes are defined
+} satisfies RouteConfig;
```
Comment (markdalgleish) [inline docs/upgrading/remix.md:121]:
```suggestion
export default [
  // however your routes are defined
] satisfies RouteConfig;
```

File: docs/upgrading/remix.md [hunk 4]
```diff
@@ -79,9 +95,127 @@ Update the scripts in your `package.json`:
...
+If you used the codemod you can skip this step as it was automatically completed.
+
+</docs-info>
+
+**👉 Add `reactRouter` plugin to `vite.config`**
+
+Change `vite.config.ts` to import and use the new `reactRouter` plugin from `@react-router/dev/vite`:
 
 ```diff
 -import { vitePlugin as remix } from "@remix-run/dev";
```
Comment (markdalgleish) [inline docs/upgrading/remix.md:221]:
```suggestion
-import { vitePlugin as remix } from "@remix-run/dev";
+import { reactRouter } from "@react-router/dev/vite";
```

File: docs/upgrading/remix.md [hunk 5]
```diff
@@ -138,9 +263,15 @@ hydrateRoot(
...
 
-### Step 7 - Update types for `AppLoadContext`
+## 8. Update types for `AppLoadContext`
+
+<docs-info>
+
+If you were using `remix-serve` you can skip this step. This is only applicable if you were using a custom server in Remix v2.
 
-<docs-info>If you were using `remix-serve` you can skip this step. This is only applicable if you were using a custom server in Remix v2.</docs-info>
+Additionally, if you used the codemod you can skip this step as it was automatically completed.
```
Comment (jrestall) [inline docs/upgrading/remix.md:272]: Codemod doesn't handle updating AppLoadContext types.

Comment (brookslybrand) [reply docs/upgrading/remix.md:272]: Whoops! Thanks for pointing that out, I'll remove it

File: docs/upgrading/remix.md (hunk 2 above)
Comment (brookslybrand) [reply docs/upgrading/remix.md]: Alright, added a comment calling this out, thanks @pcattori

**Pull request merged**